binderhub/static/js/components/NBViewerIFrame.jsx

```diff
@@ -0,0 +1,51 @@
+import { Spec } from "../spec";
+
+/**
+ * @typedef {object} NBViewerIFrameProps
+ * @prop {Spec} spec
+ * @param {NBViewerIFrameProps} props
+ * @returns
+ */
+export function NBViewerIFrame({ spec }) {
+  // We only support GitHub links as preview right now
+  if (!spec.buildSpec.startsWith("gh/")) {
+    return;
+  }
+
+  const [_, org, repo, ref] = spec.buildSpec.split("/");
+
+  let urlPath = decodeURI(spec.urlPath);
+  // Handle cases where urlPath starts with a `/`
+  urlPath = urlPath.replace(/^\//, "");
+  let filePath = "";
+  if (urlPath.startsWith("doc/tree/")) {
+    filePath = urlPath.replace(/^doc\/tree\//, "");
+  } else if (urlPath.startsWith("tree/")) {
+    filePath = urlPath.replace(/^tree\//, "");
+  }
+
+  let url;
+  if (filePath) {
+    url = `https://nbviewer.jupyter.org/github/${org}/${repo}/blob/${ref}/${filePath}`;
```

Comment:
Maybe leave a TODO note - the nbviewer host should probably someday be a config option. Doesn't need to be fixed here, since we never had this before.

Reply:
done